connectors/sources/outlook.py

```diff
@@ -0,0 +1,916 @@
+                "value": "",
+            },
+            "ssl_enabled": {
+                "depends_on": [{"field": "data_source", "value": OUTLOOK_SERVER}],
+                "display": "toggle",
+                "label": "Enable SSL",
+                "order": 10,
+                "type": "bool",
+                "value": False,
+            },
+            "ssl_ca": {
+                "depends_on": [{"field": "ssl_enabled", "value": True}],
+                "label": "SSL certificate",
+                "order": 11,
+                "type": "str",
+                "value": "",
+            },
+        }
+
+    def _pre_checks_for_get_content(
+        self, attachment_extension, attachment_name, attachment_size
+    ):
+        if attachment_extension == "":
+            self._logger.debug(
+                f"Files without extension are not supported, skipping {attachment_name}."
+            )
+            return
+
+        if attachment_extension.lower() not in TIKA_SUPPORTED_FILETYPES:
+            self._logger.debug(
+                f"Files with the extension {attachment_extension} are not supported, skipping {attachment_name}."
+            )
+            return
+
+        if attachment_size > FILE_SIZE_LIMIT:
+            self._logger.warning(
+                f"File size {attachment_size} of file {attachment_name} is larger than {FILE_SIZE_LIMIT} bytes. Discarding file content"
+            )
+            return
+        return True
+
+    async def get_content(self, attachment, timezone, timestamp=None, doit=False):
+        """Extracts the content for allowed file types.
+
+        Args:
+            attachment (dictionary): Formatted attachment document.
+            timezone (str): User timezone for _timestamp
+            timestamp (timestamp, optional): Timestamp of attachment last modified. Defaults to None.
+            doit (boolean, optional): Boolean value for whether to get content or not. Defaults to False.
+
+        Returns:
+            dictionary: Content document with _id, _timestamp and attachment content
+        """
+        attachment_size = attachment.size
+        if not (doit and attachment_size > 0):
+            return
+
+        attachment_name = attachment.name
+        attachment_extension = (
+            attachment_name[attachment_name.rfind(".") :]  # noqa
+            if "." in attachment_name
+            else ""
+        )
+
+        if not self._pre_checks_for_get_content(
+            attachment_extension=attachment_extension,
+            attachment_name=attachment_name,
+            attachment_size=attachment_size,
+        ):
+            return
+
+        self._logger.debug(f"Downloading {attachment_name}")
+
+        document = {
+            "_id": attachment.attachment_id.id,
+            "_timestamp": ews_format_to_datetime(
+                datetime=attachment.last_modified_time, timezone=timezone
+            ),
+            "_attachment": get_base64_value(attachment.content),
```

Comment:
Could `get_base64_value` block the event loop for large attachments (not sure if we are dealing just with email binary content or actual mail attachments)